src/metabase/util/schema.clj

```diff
@@ -364,17 +364,17 @@
 (def ParameterSourceOptions
   "Schema for valid source_options within a Parameter"
   ;; TODO: This should be tighter
-  {(s/optional-key :values)  [s/Str]
-   (s/optional-key :card_id) IntGreaterThanZero})
+  {(s/optional-key "values") [s/Any]
+   (s/optional-key "card_id") IntGreaterThanZero})
 
 (def Parameter
   "Schema for a valid Parameter.
   We're not using [metabase.mbql.schema/Parameter] here because this Parameter is meant to be used for
   Parameters we store on dashboard/card, and it has some difference with Parameter in MBQL."
   (with-api-error-message {:id                              NonBlankString
                            :type                            keyword-or-non-blank-str
-                           (s/optional-key :source_type)    (s/enum "static-list" "card")
-                           (s/optional-key :source_options) ParameterSourceOptions
+                           (s/optional-key :source_type)    (s/enum "static-list" "card" nil)
+                           (s/optional-key :source_options) Map
```

Comment:
I agree with adding `nil` to `source_type`, but why the other changes?

Reply:
I'm pretty sure that `values` and `card_id` come back as keywords (not strings) from the type normalization

Reply:
yep it is, I changed it to a keyword and use the paramtersourceoptions